pkg/subroutines/logger_helper.go

```diff
@@ -0,0 +1,23 @@
+package subroutines
+
+import (
+	"context"
+
+	"github.com/openmfp/golang-commons/context/keys"
+	"github.com/rs/zerolog"
+)
+
+func logInfo(ctx context.Context, msg string, err error) {
+	if ctx == nil {
```

Comment:
So the context cannot be nil in any lifecycle reconcilation. 
Also the logger is sure to be there. Checking everywere for != nil can be omitted.
The logger also has a Logr method https://github.com/openmfp/golang-commons/tree/main/logger#logr-instance
so there is no need to cast. In the end you just have to get the logger from the context once, and use it with the respective log statements